core/src/main/java/com/linecorp/armeria/client/encoding/HttpDecodedResponse.java

```diff
@@ -110,20 +111,35 @@ protected HttpObject filter(HttpObject obj) {
 
     @Override
     protected void beforeComplete(Subscriber<? super HttpObject> subscriber) {
-        if (responseDecoder == null) {
+        final HttpData lastData = closeResponseDecoder();
+        if (lastData == null) {
             return;
         }
-        final HttpData lastData = responseDecoder.finish();
         if (!lastData.isEmpty()) {
             subscriber.onNext(lastData);
         }
```

Comment:
How about releading `lastData` is it's empty? It may contain an empty direct buffer.

Reply:
Thanks all fixed. 😉